shared/src/main/java/gov/hhs/cdc/trustedintermediary/utils/SyncRetryTask.java

```diff
@@ -0,0 +1,48 @@
+package gov.hhs.cdc.trustedintermediary.utils;
+
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import java.util.concurrent.Callable;
+import javax.inject.Inject;
+
+/**
+ * Provides a reusable utility for retrying a task with a specified number of retries and wait time
+ * between retries.
+ */
+public class SyncRetryTask {
+    private static final SyncRetryTask INSTANCE = new SyncRetryTask();
+    @Inject Logger logger;
+
+    public static SyncRetryTask getInstance() {
+        return INSTANCE;
+    }
+
+    private SyncRetryTask() {}
+
+    public <T> boolean retry(Callable<T> task, int maxRetries, long waitTime) {
```

Comment:
I recommend that we return `T` instead of boolean and that this method throws `Exception` that will originate from the `Callable` on the last retry failure.

This get's us two benefits.
1. We get the return value back from the `Callable` if we pass in one that actually returns something.  E.g. `Callable<Integer>`.
2. More importantly, we maintain the Exception cause chain.  If an Exception is thrown inside the `Callable`, that eventually results in a failure, we will want to have that stacktrace of that exception.  The way the code is written today, we loose that cause chain by always catching the Exception and eventually returning `false` for a failure.

What do you think?

Reply:
Those are great points, and I agree that's a better implementation. I'll make the changes